frontend/src/components/NewRuleForm.vue

```diff
@@ -26,6 +26,13 @@ const { mutateAsync } = useAddRuleMutation();
 const trackingRuleName = ref("");
 
 const generationRulesCount = ref(0);
+
+// Dirty state to track unsaved changes
+const isDirty = ref(false)
+const markDirty = () => {
+  isDirty.value = true
+}
+
```

Comment:
You don't actually need to track the dirtiness manually, FormKit does it automatically. You can use its state in the `context` property. For example:
```suggestion
const newRuleForm = ref<{ node: FormKitNode } | null>(null);
const isDirty = computed(
  () => newRuleForm.value?.node.context?.state.dirty === true,
);
```

You can then remove `markDirty` everywhere.